### PR TITLE
refactor: remove some dead code

### PR DESCRIPTION
**Pull request by e-dard**
Needs #1025.

This is a follow on from #1025 that removes some unnecessary code and also implements a refactor suggestion of @carols10cents's. 

- [x] Remove some dead code. 
- [ ]  Implement First/Last aggregates
- [ ]  Implement windowed aggregates
- [ ]  Refactor row_group.rs to move code into modules as the file is getting way too big now.

### CODE REVIEW

**Comment by NGA-TRAN** [comment]
Nice to see a lot of dead close removed and errors are caught and thrown

File: read_buffer/src/chunk.rs [hunk 1]
```diff
@@ -205,17 +205,20 @@ impl Chunk {
...
-        // TODO(edd): this should return an error
-        chunk_data
+        let table = chunk_data
             .data
             .get(table_name)
-            .map(|table| table.read_aggregate(predicate, group_columns, aggregates))
+            .ok_or(Error::TableNotFound {
+                table_name: table_name.to_owned(),
+            })?;
+
```
Comment (alamb) [inline read_buffer/src/chunk.rs:218]: I think you can use the `OptionExt` trait to make this:

```suggestion
            .context(TableNotFound {table_name})?;

```

**Review by alamb — APPROVED** [review]
seems like a nice cleanup to me -- I have to defer to you if this code is needed or not but it seems reasonable to me to delete

File: read_buffer/src/chunk.rs (hunk 1 above)
Comment (e-dard) [reply read_buffer/src/chunk.rs]: Fixed this in a couple of places in 585213e. Thanks.

**Pull request merged**